### PR TITLE
Arcconf Plugin patch: Fix for 'Pools' function

### PR DESCRIPTION
**Pull request by sharath-t**
issue: pool ID for array with multiple LDs - number of pools shown is incorrect. current logic is "ctrl ID:LD ID"
		fix: replaced existing logic with logic to read arrayID
		
issue: total space and free space is displayed as 0
		fix: implemented logic for 'total space = blockSize * totalSize' and 'free space = blockSize * unUsedSpace'

CI kick

### CODE REVIEW

File: plugin/arcconf/arcconf.py [hunk 1]
```diff
@@ -398,22 +397,24 @@ def pools(self, search_key=None, search_value=None,
...
+            if 'Array' in decoded_json['Controller']:
+                array_infos = decoded_json['Controller']['Array']
+                num_array = len(array_infos)
+                for array in range(num_array):
+                    array_id = array_infos[array]['arrayID']
+                    array_name = array_infos[array]['arrayName']
+                    block_size = array_infos[array]['blockSize']
+                    total_size = array_infos[array]['totalSize']
+                    unused_size = array_infos[array]['unUsedSpace']
+                    lsm_pools.append(Arcconf._arcconf_array_to_lsm_pool(
```
Comment (cathay4t) [inline plugin/arcconf/arcconf.py]: Can you pass the `array` as single argument to `_arcconf_array_to_lsm_pool()`?

Like:
```python
def _arcconf_array_to_lsm_pool(arcconf_array):
    array_id = arcconf_array['arrayID']
    array_name = arcconf_array['arrayName']
    block_size = arcconf_array['blockSize']
    total_size = arcconf_array['totalSize']
    unused_size = arcconf_array['unUsedSpace']
    ...


lsm_pools.append(Arcconf._arcconf_array_to_lsm_pool(array_infos[array])

```

File: plugin/arcconf/arcconf.py [hunk 2]
```diff
@@ -375,18 +375,17 @@ def systems(self, flags=0):
...
-        free_space = 0
-        total_space = 0
-        # TODO Need to have a logic for finding free space and total space.
+        free_space = int(block_size) * int(unused_size)
+        total_space = int(block_size) * int(total_size)
 
         status = Pool.STATUS_OK
         status_info = ''
-        plugin_data = "%s:%s" % (ctrl_num, ld_name)
+        plugin_data = {'ctrl_id': sys_id, 'array_id': array_id}
```
Comment (cathay4t) [inline plugin/arcconf/arcconf.py]: The `plugin_data` should be a `string` rather than a `dict`. Just use the `pool_id` would be enough.

**Comment by sharath-t** [comment]
@cathay4t Thanks for taking time in reviewing.

can you help me in understanding why 'plugin_data' should be string only? i have used it as dictionary to ease of access in volume dependent functions.

meanwhile, i will start refactoring code to address the comments

**Comment by cathay4t** [comment]
> @cathay4t Thanks for taking time in reviewing.
> 
> can you help me in understanding why 'plugin_data' should be string only? i have used it as dictionary to ease of access in volume dependent functions.

The libstoragemgmt also provide C interface, in order to support it we define `plugin_data` as string
quit while ago. Since we have ABI to maintain, all plugin should enforce that.

File: plugin/arcconf/arcconf.py (hunk 1 above)
Comment (sharath-t) [reply plugin/arcconf/arcconf.py]: Done - commit id - e5c44368c28265b40026398ae820e2b6ad6bcdb8

File: plugin/arcconf/arcconf.py (hunk 2 above)
Comment (sharath-t) [reply plugin/arcconf/arcconf.py]: Done - commit id - e5c4436

**Comment by sharath-t** [comment]
i see that all other PRs would fail as there is no support for volume-raid-create-cap. 
@cathay4t if you can help me merge https://github.com/libstorage/libstoragemgmt/pull/381 then i can run CI on rest of the PRs.

Thanks.

**Comment by sharath-t** [comment]
After the merge of PR #381 . i may have used "Squash and merge" option on "sharath-t:sharath-t-patch-3" branch and it has created a merge commit for this PR as well.

i should have used "Rebase and merge" on "sharath-t:sharath-t-patch-3".

is it possible to cherry-pick the first two commits and merge the PR? or should i revert the merged commit and rebase?

**Comment by tasleson** [comment]
Integrated into https://github.com/libstorage/libstoragemgmt/pull/386

**Pull request closed without merging**